### PR TITLE
Rework @gear-api docs

### PR DESCRIPTION
**Pull request by EugenWay**
Updated sections:

-
-
-

### CODE REVIEW

File: docs/api/calculate-gas.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# Calculate gas
+
+Any operations of the program code execution or processing of messages are paid with gas by the initiator of any of these actions.
+
+To guarantee successful message processing and avoid errors like `Gaslimit exceeded`, you can simulate the execution in advance and calculate the exact value of the gas consumed.
+
+
+## Calculate gas for messages
+
+Depending on the conditions, you can calculate the gas for initializing the program, processing the message in handle() or reply()
```
Comment (shamilsan) [inline docs/api/calculate-gas.md:15]:
```suggestion
Depending on the conditions, you can calculate the gas for initializing the program, processing the message in `handle()` or `reply()`.
```

File: docs/api/calculate-gas.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+### Init message
+
+```javascript
+// get program code
+const code = fs.readFileSync('demo_ping.opt.wasm');
+const gas = await gearApi.program.calculateGas.init(
+  '0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d', // source id
+  code,
+  '0x00', // payload
+  0, //value
```
Comment (shamilsan) [inline docs/api/calculate-gas.md:34]:
```suggestion
  0, // value
```

File: docs/api/calculate-gas.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+console.log(gas.toHuman());
+```
+
+### Handle 
+
+```javascript
+const meta = await getWasmMetadata(fs.readFileSync('demo_meta.opt.wasm'));
+const estimatedGas = await gearApi.program.calculateGas.handle(
+  '0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d', // source id
+  '0xa178362715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d', //program id
```
Comment (shamilsan) [inline docs/api/calculate-gas.md:46]:
```suggestion
  '0xa178362715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d', // program id
```

File: docs/api/extra-queries.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
+---
+sidebar_position: 10
+sidebar_label: Cookbook
+---
+
+# Cookbook
+
+Some batch of useful code snippets in a question-answer format:
+
+##Subscribe to new blocks
```
Comment (shamilsan) [inline docs/api/extra-queries.md:10]:
```suggestion
## Subscribe to new blocks
```

File: docs/api/read-state.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+The following query is used to read the state of the program:
+
+```javascript
+const metaWasm = fs.readFileSync('path/to/meta.wasm');
+const state = await gearApi.programState.read(programId, metaWasm, inputValue?);
+```
+`programId` is a payload If program expects it in `meta_state`
+
+## cookbook
```
Comment (shamilsan) [inline docs/api/read-state.md:16]:
```suggestion
## Cookbook
```

File: docs/api/read-state.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+# Read State
+
+The following query is used to read the state of the program:
+
+```javascript
+const metaWasm = fs.readFileSync('path/to/meta.wasm');
+const state = await gearApi.programState.read(programId, metaWasm, inputValue?);
+```
+`programId` is a payload If program expects it in `meta_state`
```
Comment (shamilsan) [inline docs/api/read-state.md:14]:
```suggestion
`programId` is a payload if program expects it in `meta_state`
```

File: docs/api/submit-code.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+---
+sidebar_position: 8
+sidebar_label: Submit Code
+---
+
+# Submit Code
+
+If need to load the program code into the chain without initialization, we can look at the following example:
```
Comment (shamilsan) [inline docs/api/submit-code.md:8]:
```suggestion
If we need to load the program code into the chain without initialization, we can look at the following example:
```

File: docs/api/tooling/meta-cli.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+<b>encode</b> - _Encode payload to hex_
+<b>meta</b> - _Display metadata from .meta.wasm_
+<b>type</b> - _Display type structure for particular type from .meta.wasm_
+
+You can simply run these commands and you will be prompted to enter the neccessary data. Or you can specify data through options:
+
+<b>-t, --type</b> - _Type to encode or decode the payload. If it will not specified you can select it later_
+<b>-m, --meta</b> - _Path to .meta.wasm file with program's metadata_
+<b>-o --output</b> - _Output json file. If it doesn't exist it will be created_
+<b>-j --payloadFromJson</b> - _If need to take the payload from json_
```
Comment (shamilsan) [inline docs/api/tooling/meta-cli.md:42]: It works bad, see: https://deploy-preview-205--competent-kepler-ce8c2a.netlify.app/api/tooling/meta-cli#available-commands

You may use asterisks for `**bold**`.

**Review by AndrePanin — APPROVED** [review]

File: docs/api/tooling/meta-cli.md (hunk 8 above)
Comment (AndrePanin) [reply docs/api/tooling/meta-cli.md]: fixed

**Pull request merged**